tests/unit/algos/list/delimiterMaster.spec.js

```diff
@@ -46,6 +47,99 @@ function getListingKey(key, vFormat) {
+        gteparams: null,
+    }
+}
+
+/**
+ * Simplified version of logic found in Metadata's RepdServer. When MAX_STREAK_LENGTH (typically 100)
+ * is reached, we attempt to skip to the next listing range as a performance optimization. 
+ * @param {Integer} filteringResult - 0, 1, -1 indicating if we can skip to the next character range.
+ * @param {String} skippingRange - lower bound that the listing can begin from.
+ * @returns 
+ */
+function handleStreak(filteringResult, skippingRange, state) {
+    if (filteringResult < 0) {
+        // readStream would be destroyed. In this mock, we just continue.
+    } else if (filteringResult === 0 && skippingRange) {
+        // check if MAX_STREAK_LENGTH consecutive keys have been
+        // skipped
+        if (++state.streakLength === MAX_STREAK_LENGTH) {
+            if (Array.isArray(skippingRange)) {
+                // With synchronized listing, the listing
+                // algo backend skipping() function must
+                // return as many skip keys as listing
+                // param sets.
+                for (let i = 0; i < skippingRange.length; ++i) {
+                    // eslint-disable-next-line no-param-reassign
+                    state.params[i].gte = inc(skippingRange[i]);
+                }
+            } else {
+                // eslint-disable-next-line no-param-reassign
+                
+                state.params.gte = inc(skippingRange);
+            }
+            if (state.gteparams && state.gteparams === state.params.gte) {
+                streakLength = 1;
+            } else {
+                // stop listing this key range
+                // eslint-disable-next-line no-param-reassign
+                state.gteparams = state.params.gte;
+            }
+        }
+    } else {
+        state.streakLength = 1;
+    }
+}
+
+/**
+ * Generate a random number of versioned keys.
+ * @param {Integer} maxKeys - how many versioned keys to generate
+ * @yields {Object} - { key, isDeleteMarker }
+ */
+function *generateVersionedKeys(masterKey, numKeys) {
+    for (let i = 0; i < numKeys; i++) {
+        yield {
+            key: `${masterKey}${VID_SEP}${zpad(i)}`, 
+            isDeleteMarker: chance.bool(),
+            canSkip: true
+        }
+    }
+}
+
+/**
+ * Generate raw listing keys in alphabetical order.
+ * @param {Integer} number - how many keys to generate.
+ * @yields {Object} - { key, isDeleteMarker }
+ */
+
+function *generateKeys(count) {
+    let idx = 0;
+    let masterKey = '_Common-Prefix/';
+    yield { key: masterKey, isDeleteMarker: false, canSkip: false }
+
+    idx++;
+    while (idx < count) {
+        masterKey = `_Common-Prefix/${zpad(idx)}`;
+        const masterKeyWithSubprefix = `_Common-Prefix/${zpad(idx)}/${zpad(idx)}`
+        const baseKey = chance.bool() ? masterKeyWithSubprefix : masterKey;
+        const isDeleteMarker = chance.bool();
+        yield { key: baseKey, isDeleteMarker, canSkip: isDeleteMarker}
```

Comment:
I am not sure the `canSkip` value is right here: we should be able to skip keys in subprefixes if we have already seen the same subprefix. I think the current code is doing this (in delimiter.js `addCommonPrefix`) but the test does not seem to reflect this.

Reply:
good catch